### DISCUSSION TITLE
Make Installation Error - v11.1.0

### DISCUSSION THREAD

**Discussion by hootmet** [dtcenter/METplus#2617] [Installation] [answered]
Met installation goes well right down to what literally appears to be the last step.

**cli comand:**
 ./compile_MET_all.sh install_met_env.jupyter-env
 
 **End of compile script:**
 configure_cmd="${configure_cmd} ${OPT_ARGS}"
 
echo "cd `pwd`"
run_cmd "${configure_cmd} > configure.log 2>&1"
run_cmd "make ${MAKE_ARGS} > met.make.log 2>&1"
run_cmd "make install > met.make_install.log 2>&1"
run_cmd "make test > met.make_test.log 2>&1"
 
echo "Finished compiling at `date`"

**Error message:**
make test > met.make_test.log 2>&1
ERROR: Command returned with non-zero (2) status: make test > met.make_test.log 2>&1

**Log output:**
cat  ./external_libs/netcdf/netcdf-cxx4-4.3.1/met.make_test.log
make: *** No rule to make target 'test'.  Stop.

**Comment by jprestop** [comment]
@hootmet Could you please provide the following files for us to review?

1. The entire screen output from the run of the script
2. Your install_met_env.jupyter-env file
3. The config.log file (not the configure.log file) from the MET directory
4. The met.make.log file from the MET directory
5. The met.make_test.log file from the MET directory

Please feel free to either attach them via GitHub Discussions here in this post or provide them via FTP server as described [here](https://github.com/dtcenter/METplus/discussions/954).

**Reply by hootmet** [reply]
Attached find the requested information.
[requested_info.zip](https://github.com/user-attachments/files/15917020/requested_info.zip)


Thank you for the help

**Reply by jprestop** [reply]
@hootmet Thank you for providing these files.  It was helpful to debug the problem.  In your met_screen.txt file, I see the following:

```
Compiling MET at Tue Jun 18 22:56:15 UTC 2024
tar (child): /efs/share/jhthomps/tar_files/v11.1.0.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
./compile_MET_all.sh: line 671: cd: /efs/share/jhthomps/MET*: No such file or directory
```

It looks like you perhaps you have not downloaded the [v11.1.0.tar.gz](https://github.com/dtcenter/MET/archive/refs/tags/v11.1.0.tar.gz) file and placed it in the expected `tar_files` subdirectory. Please download the file and place it in a `tar_files` directory that is at the same level as the `compile_MET_all.sh` script (The `/efs/share/jhthomps/tar_files` directory from what I can tell from your configuration file).  Then, try running the script again.  Please let us know how it goes.

**Comment by hootmet** [comment]
Well that certainly helped but now errors out per the attached file....

[met.make.log](https://github.com/user-attachments/files/15928735/met.make.log)

**Reply by jprestop** [reply]
Thank you @hootmet.  Could you also please pass along the config.log file located in the same directory as your met.make.log file?

**Comment by hootmet** [comment]
here you go....

[config.log](https://github.com/user-attachments/files/15931298/config.log)

**Reply by jprestop** [reply]
Thank you @hootmet.  In your install_met_env file, please try changing the value of MET_PYTHON_CC to 
`-I/srv/conda/envs/notebook/include/python3.1`.  Then, try recompiling and let us know how it goes.

**Comment by hootmet** [comment]
Failed in the Make Met section with ....

python_line.h:23:10: fatal error: Python.h: No such file or directory

**Reply by jprestop** [reply]
@hootmet Thank you.  I see I had a typo in what I told you to set MET_PYTHON_CC to.  Instead of setting it to
`-I/srv/conda/envs/notebook/include/python3.1`
it should have been
`-I/srv/conda/envs/notebook/include/python3.11`.

Please give that a try.

**Comment by hootmet** [comment]
Still issues, see attached....

[met.make.log](https://github.com/user-attachments/files/15960271/met.make.log)

**Reply by jprestop** [reply]
Thank you @hootmet.  I'm sorry to see you're still having trouble.  Could you please also pass along your config.log file from the same directory as your met.make.log file?

**Comment by hootmet** [comment]
[config.log](https://github.com/user-attachments/files/15960900/config.log)
[configure.log](https://github.com/user-attachments/files/15960901/configure.log)

**Comment by hootmet** [comment]
Any feedback on logs previously sent?

**Reply by jprestop** [reply, marked as answer]
We haven't encountered this problem before, but we have had users encounter similar problems.  I believe the problem you are experiencing is likely due to a mismatch in compilers in /usr/bin and the conda environment.  I see the following in your config.log file:
```
CC='/usr/bin/gcc'
CPP='/srv/conda/envs/notebook/bin/x86_64-conda-linux-gnu-cpp'
CXX='/usr/bin/c++'
```
In your install_met_env.jupyter-mvu file, please try setting the following:
```
export CPP=/usr/bin/c++
```
as '/srv/conda/envs/notebook/bin/x86_64-conda-linux-gnu-cpp' is currently being used, but we don't want it to be used.

I also see that the following is being set in your install_met_env.jupyter-mvu file:
```
export COMPILER=gnu_12.3.0
```
but in your config.log file it shows that the following compiler is actually being used:
```
gcc version 11.4.0
```
so please update your install_met_env.jupyter-mvu file to:
```
export COMPILER=gnu_11.4.0
```
and try recompiling.  Please let us know how it goes.

**Comment by hootmet** [comment]
No joy I'm afraid, see log.

Thanks for all the help. It's much appreciated.
[gsl.configure.log](https://github.com/user-attachments/files/15993613/gsl.configure.log)

**Reply by jprestop** [reply]
Hi @hootmet.  I'm sorry to hear that.  If GSL did not compile, could you please pass along the output from the compilation script along with the config.log file in the GSL directory?  Unfortunately, the configure.log files don't contain any useful information.  They simply say at the end `See 'config.log' for more details`.

**Comment by hootmet** [comment]
Here you go


[compile_06262024.txt](https://github.com/user-attachments/files/15994192/compile_06262024.txt)
[config.log](https://github.com/user-attachments/files/15994193/config.log)

**Reply by jprestop** [reply]
Thanks @hootmet.  Can you please add the following line to the end of your install_met_env.jupyter-mvu file and send me the output?
```
printenv
```

**Comment by hootmet** [comment]
here you go...

[printenv.txt](https://github.com/user-attachments/files/15994764/printenv.txt)

**Reply by jprestop** [reply]
@hootmet Thank you.  Can you please tell me - does `cpp` exist in `/usr/bin`?

**Comment by hootmet** [comment]
(notebook) hootmet@jupyter-hootmet:/usr/bin$ ls /usr/bin | grep cpp
cpp
cpp-11
x86_64-linux-gnu-cpp
x86_64-linux-gnu-cpp-11

**Reply by jprestop** [reply]
Great, @hootmet!
Please update your install_met_env.jupyter-mvu file to be:
```
export CPP=/usr/bin/cpp
```
and rerun.

**Comment by hootmet** [comment]
Failed in the make MET section again.....

/srv/conda/envs/notebook/bin/ld: /srv/conda/envs/notebook/bin/../x86_64-conda-linux-gnu/sysroot/lib64/[libdl.so](http://libdl.so/).2: undefined reference to `_dl_addr@GLIBC_PRIVATE'
/srv/conda/envs/notebook/bin/ld: /srv/conda/envs/notebook/bin/../x86_64-conda-linux-gnu/sysroot/lib64/[libdl.so](http://libdl.so/).2: undefined reference to `_dl_vsym@GLIBC_PRIVATE'
/srv/conda/envs/notebook/bin/ld: /srv/conda/envs/notebook/bin/../x86_64-conda-linux-gnu/sysroot/lib64/[libpthread.so](http://libpthread.so/).0: undefined reference to `__libc_pthread_init@GLIBC_PRIVATE'
/srv/conda/envs/notebook/bin/ld: /srv/conda/envs/notebook/bin/../x86_64-conda-linux-gnu/sysroot/lib64/[libpthread.so](http://libpthread.so/).0: undefined reference to `_dl_make_stack_executable@GLIBC_PRIVATE'
collect2: error: ld returned 1 exit status
make[5]: *** [Makefile:436: ensemble_stat] Error 1
make[5]: Leaving directory '/efs/share/jhthomps/MET-11.1.0/src/tools/core/ensemble_stat'
make[4]: *** [Makefile:381: all-recursive] Error 1
make[4]: Leaving directory '/efs/share/jhthomps/MET-11.1.0/src/tools/core'
make[3]: *** [Makefile:375: all-recursive] Error 1
make[3]: Leaving directory '/efs/share/jhthomps/MET-11.1.0/src/tools'
make[2]: *** [Makefile:372: all-recursive] Error 1
make[2]: Leaving directory '/efs/share/jhthomps/MET-11.1.0/src'
make[1]: *** [Makefile:429: all-recursive] Error 1
make[1]: Leaving directory '/efs/share/jhthomps/MET-11.1.0'
make: *** [Makefile:370: all] Error 2

**Reply by jprestop** [reply]
@hootmet Can you please send the config.log file and the met.make.log file?

**Comment by hootmet** [comment]
[config.log](https://github.com/user-attachments/files/15996446/config.log)
[met.make.log](https://github.com/user-attachments/files/15996447/met.make.log)

**Reply by jprestop** [reply]
@hootmet it looks like that is the config.log file for GSL.  Could you please send the config.log file for MET?  It will be in the same directory as the met.make.log file.

**Reply by jprestop** [reply]
@hootmet Actually, my apologies.  It was the correct file.  I was looking at an older file.  Sorry about that!

**Reply by jprestop** [reply]
Thanks @hootmet.  I see the following in your printenv.txt file:
```
PATH=/srv/conda/envs/notebook/bin:/srv/conda/condabin:/srv/conda/envs/notebook/bin:/srv/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/goofys/bin
```
Please add
```
export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/goofys/bin:/srv/conda/envs/notebook/bin:/srv/conda/condabin:/srv/conda/envs/notebook/bin:/srv/conda/bin:
```
above 
```
export GCC=/usr/bin/gcc
```
to prioritize the system level executables in your install_met_env.jupyter-mvu.  

Before rerunning, there's no need for you to recompile all of the external libraries again, so please change
```
#export EXTERNAL_LIBS=${TEST_BASE}/external_libs
#export MET_GSL=${EXTERNAL_LIBS}
#export MET_BUFRLIB=${EXTERNAL_LIBS}/lib
#export BUFRLIB_NAME=-lbufr
#export LIB_JASPER=${EXTERNAL_LIBS}/lib
#export LIB_LIBPNG=${EXTERNAL_LIBS}/lib
#export LIB_Z=${EXTERNAL_LIBS}/lib
#export MET_GRIB2CLIB=${EXTERNAL_LIBS}/lib
#export MET_GRIB2CINC=${EXTERNAL_LIBS}/include
#export GRIB2CLIB_NAME=-lgrib2c
#export MET_HDF5=${EXTERNAL_LIBS}
#export MET_NETCDF=${EXTERNAL_LIBS}
```
in your install_met_env.jupyter-mvu to be
```
export EXTERNAL_LIBS=${TEST_BASE}/external_libs
export MET_GSL=${EXTERNAL_LIBS}
export MET_BUFRLIB=${EXTERNAL_LIBS}/lib
export BUFRLIB_NAME=-lbufr
export LIB_JASPER=${EXTERNAL_LIBS}/lib
export LIB_LIBPNG=${EXTERNAL_LIBS}/lib
export LIB_Z=${EXTERNAL_LIBS}/lib
export MET_GRIB2CLIB=${EXTERNAL_LIBS}/lib
export MET_GRIB2CINC=${EXTERNAL_LIBS}/include
export GRIB2CLIB_NAME=-lgrib2c
export MET_HDF5=${EXTERNAL_LIBS}
export MET_NETCDF=${EXTERNAL_LIBS}
```
That way, the script will only recompile MET.  Please let us know how it goes.

**Comment by hootmet** [comment]
And there was joy throughout the land!!!

It seemed to compile cleanly. Is there a quick test that can be run?

**Comment by hootmet** [comment]
Actually, we found the test scripts. All looks good!

Thank you very much for your incredible help, support and patience.

**Reply by jprestop** [reply]
That's great news @hootmet!  Thank you for letting us know.  I'll go ahead and select an answer and mark this conversation as locked to prevent future posts here.  Please feel free to start a new discussion with any new questions.
